proposals/0011-inline-spirv.md

```diff
@@ -0,0 +1,293 @@
+
+### Builtin output
+
+The existing inline SPIR-V has limited support for adding a builtin output. This
+would have most of the same problems as declaring a builtin input.
+
+To support adding builtin outputs from source, this proposal adds a new
+`vk::ext_builtin_output attribute` that applies to a function declaration
+providing an implicit definition which sets the value of the output to the
+provided parameter value.
+
+For example, the `gl_FragStencilRefARB` builtin could be declared in a header
+file like this:
+
+```
+[[vk::ext_extension("SPV_EXT_shader_stencil_export")]]
+[[vk::ext_builtin_input(/* FragStencilRefEXT */ 5014)]]
+void gl_FragStencilRefARB(int);
+```
+
+Then the compiler will be able to add a variable in the output storage class,
+with the builtin decoration, with a type that is the same as the parameter, and
+add it to the OpEntryPoint for the entry points from which it is reachable.
+
+The developer can set the builtin output by simply calling the function.
+
+### Capability and extension instructions
+
+Existing inline SPIR-V has attributes to indicate that a capability or extension
+is needed by a particular part of the code. There are examples above. This
+proposal will allows these attribute to be applied to an entry point.
+
+## Detailed design
+
+## Alternatives considered (Optional)
+
+### Auto-generating builtin functions and attributes from [spirv.core.grammar.json](https://github.com/KhronosGroup/SPIRV-Headers/blob/8e2ad27488ed2f87c068c01a8f5e8979f7086405/include/spirv/unified1/spirv.core.grammar.json)
+
+When we considered auto-generating, we noticed that not enough information is
+available. For example, it is not possible to tell if a builtin should be an
+input or output, and we cannot know its types. The other problem is that it will
+require testing in DXC when new functions or attributes are added. The solution
+proposed above leave a smaller testing surface in DXC, and places the testing
+burden on the writer of the header file.
+
+### Removing explicit references to capabilities and extensions
+
+It is possible to get the list of required capabilities from
+[spirv.core.grammar.json](https://github.com/KhronosGroup/SPIRV-Headers/blob/8e2ad27488ed2f87c068c01a8f5e8979f7086405/include/spirv/unified1/spirv.core.grammar.json).
+It would generally be possible for the compiler to automatically add the
+required capabilities an instruction is used. However, there are some cases
+where the capability allows an instruction to take a new type of operand. These
+are not part of the grammar, and would still require some way of explicitly
+adding a capability and annotation.
+
+If we can hide all of the cases that could be implicitly added in a header file,
+then there is very little burden on users. This is why we chose to not have
+implicit inclusion of the capabilities and extensions.
+
+## Acknowledgments (Optional)
+
+Take a moment to acknowledge the contributions of people other than the author
+and sponsor.
```

Comment:
I think this section can be removed instead of leaving the boilerplate around if there aren't any specific acknowledgements